techTree.md

```diff
@@ -0,0 +1,97 @@
+
+The Ethereum state is 40-100GB, depending on how you store it. The de facto network primitive for state sync is `getNodeData`, which requests the state trie node by node from a pool of (full node) peers. This is a slow and inefficient method of sync that puts a heavy burden on fully synced nodes serving greedy leechers. 
+
+An alternative network primitive is sought that can eliminate or at least substantially reduce reliance on `getNodeData` for state sync. 
+
+#### SNAP
+
+A promising auxilary protocol meant to be run side-by-side with the eth protocol, not standalone (e.g. chain progression is announced via eth). The `snap` protocol permits downloading the entire Ethereum state without having to download all the intermediate Merkle proofs, which can be regenerated locally.
+
+[The snap protocol](https://github.com/ethereum/devp2p/blob/3fe9713658f3b3b56e4e99493c54f313e11b43a0/caps/snap.md) is currently in an early version, but may be the most promising alternative to current sync strategies, and a viable path to improving state sync
+
+#### Merry-go-round
+
+At a high level, [MGR](https://ethresear.ch/t/merry-go-round-sync/7158) would operate by enumerating the full state in a predetermined order and gossiping this data among the clients which are actively syncing. For a client to fully sync it needs to “ride” one full rotation of the merry-go-round. There has been some discussion around the best way to organize the state such that it can be 'rotated' through, as well as how storage tries and code blocks will be handled when breaking the full state into several similarly sized pieces for sync.
+
+#### Beam sync
+
+While not truly part of 'Stateless Ethereum', beam sync as implemented in Trinity and Nethermind is 'proto-stateless' in that clients without state try to execute a block *before* they have the complete state, and request missing pieces 'on the fly' from peers. Beam sync is a foot-in-the-door for stateless clients and a witness prototype, an beam syncing nodes could provide much needed experimental data to help with other research. 
+
+
+
+## Binary State Trie Transition
+
+In theory converting the existing [hexary patricia trie] to a binary format could reduce the size of a witness substantially, and this has been [supported by early research](https://medium.com/@mandrigin/stateless-ethereum-binary-tries-experiment-b2c035497768). Re-computing the Ethereum state as a binary trie is a complex transition to manage, and there are at the moment a few proposals for effecting that change. 
+
+#### Overlay method
+
+Although there are a few abstract strategies for transitioning to a binary trie, the [one that is currently being worked](https://medium.com/@gballet/ethereum-state-tree-format-change-using-an-overlay-e0862d1bf201) on entails a transition period in which both tries (hexary and binary) are computed in parallel by full nodes, and only after the entire state has been converted and verified is the legacy structure abandoned. 
```

Comment:
"computed in parallel" is vague. I believe that the hexary tree becomes read-only as it is converted to binary tree, and only the overlay tree is updated. Once the read-only hexary to binary tree transition is completed, the overlay tree becomes read-and-delete-only as nodes are deleted from it and written to the binary tree.

Reply:
mmmm okay, let me see if I can re-write this because I admittedly just got lazy and figured better to be vague than wrong :sweat_smile: